op/framework.py

```diff
@@ -497,7 +497,16 @@ class SomeObject:
             if not hasattr(observer, method_name):
                 raise RuntimeError(f'Observer method not provided explicitly and {type(observer).__name__} type has no "{method_name}" method')
 
-        # TODO Validate that the method has the right signature here.
+        # Validate that the method has an acceptable call signature.
+        sig = inspect.signature(getattr(observer, method_name))
+        if len(sig.parameters) < 1:  # Note: self isn't included.
+            raise TypeError(f'{type(observer).__name__}.{method_name} must accept event parameter')
+        elif len(sig.parameters) > 1:
+            # Allow for additional optional params, since there's no reason to exclude them. Note: we only
+            # need to check the second, because once you start allowing optional params, you have to continue.
```

Comment:
Probably not worth catching as it's an unusual construct, but FWIW:
```
>>> def f(a, b=1, *, c): pass                                                                                                                                            
...                                                                                                                                                                      
>>> f(1)                                                                                                                                                                 
Traceback (most recent call last):                                                                                                                                       
  File "<stdin>", line 1, in <module>
TypeError: f() missing 1 required keyword-only argument: 'c'
```

Let's please drop the "Note: " though (all comments are "please note this"). We may as well drop the second phrase entirely, as it's not entirely true.